framework/db/BaseActiveRecord.php

```diff
@@ -616,6 +617,30 @@ public function isAttributeChanged($name, $identical = true)
         return isset($this->_attributes[$name]) || isset($this->_oldAttributes[$name]);
     }
 
+    /**
+     * Returns a value indicating whether the model has at least one attribute dirty.
+     * @return bool whether the model has at least one attribute dirty
+     * @since 3.0.0
+     */
+    public function isDirty()
+    {
+        if ($this->_attributes === [] and $this->attributes() === []) {
+            return false;
+        }
+
+        if ($this->_oldAttributes === null) {
+            return true;
+        }
+
+        foreach ($this->_attributes as $name => $value) {
+            if (!array_key_exists($name, $this->_oldAttributes) || $value !== $this->_oldAttributes[$name]) {
```

Comment:
Maybe it's better to use `isAttributeChanged` here?

Reply:
Maybe, but the first reason was to avoid overhead (in term of performance) and because I didn't like the code and it's not equivalent. I could also use `getIsNewRecord()` one line above.

However, now that you mentioned it, I feel like `isAttributeChanged()` or here `isAttributeDirty()` is not coded properly and is bugged. I will submit a PR and add tests for it too.

Reply:
I think you should use `getIsNewRecord()`. We should optimize for performance for relative stuff, function calls are really fast in PHP and optimizing by code duplication is not worth it IMO.

Also it breaks logic, I can override `getIsNewRecord()` and would expect `isDirty` to always return `true` for new records even when I use some other method of defining a new records...